content/en/docs/Examples/embedded_images.md

```diff
@@ -0,0 +1,464 @@
+     - name: "Setup groups"
+       ensure_entities:
+       - entity: |
+            kind: "group"
+            group_name: "admin"
+            password: "x"
+            gid: 900
+     # Setup network - openSUSE specific
+     - name: "Network setup"
+       files:
+       - path: /etc/sysconfig/network/ifcfg-eth0
+         content: |
+                  BOOTPROTO='dhcp'
+                  STARTMODE='onboot'
+         permissions: 0600
+         owner: 0
+         group: 0
+     # Setup a custom user
+     - name: "Setup users"
+       users:
+       # Replace the default user name here and settings
+        joe:
+          # Comment passwd for no password
+          passwd: "joe"
+          shell: /bin/bash
+          homedir: "/home/joe"
+          groups:
+          - "admin"
+       #authorized_keys:
+       # Replace here with your ssh keys
+       # joe: 
+       # - ssh-rsa ....
+     # Setup sudo
+     - name: "Setup sudo"
+       files:
+       - path: "/etc/sudoers"
+         owner: 0
+         group: 0
+         permsisions: 0600
+         content: |
+            Defaults always_set_home
+            Defaults secure_path="/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/bin:/usr/local/sbin"
+            Defaults env_reset
+            Defaults env_keep = "LANG LC_ADDRESS LC_CTYPE LC_COLLATE LC_IDENTIFICATION LC_MEASUREMENT LC_MESSAGES LC_MONETARY LC_NAME LC_NUMERIC LC_PAPER LC_TELEPHONE LC_ATIME LC_ALL LANGUAGE LINGUAS XDG_SESSION_COOKIE"
+            Defaults !insults
+            root ALL=(ALL) ALL
+            %admin ALL=(ALL) NOPASSWD: ALL
+            @includedir /etc/sudoers.d
+       commands:
+       - passwd -l root
+   # Setup persistency so k3s works properly
+   # See also: https://rancher-sandbox.github.io/cos-toolkit-docs/docs/reference/immutable_rootfs/#configuration-with-an-environment-file
+   rootfs.after:
+    - name: "Immutable Layout configuration"
+      environment_file: /run/cos/cos-layout.env
+      environment:
+        VOLUMES: "LABEL=COS_OEM:/oem LABEL=COS_PERSISTENT:/usr/local"
+        OVERLAY: "tmpfs:25%"
+        RW_PATHS: "/var /etc /srv"
+        PERSISTENT_STATE_PATHS: >-
+          /etc/systemd
+          /etc/rancher
+          /etc/ssh
+          /etc/iscsi 
+          /etc/cni
+          /home
+          /opt
+          /root
+          /usr/libexec
+          /var/log
+          /var/lib/rancher
+          /var/lib/kubelet
+          /var/lib/wicked
+          /var/lib/longhorn
+          /var/lib/cni
+        PERSISTENT_STATE_BIND: "true"
+   # Finally, let's start k3s when network is available, and download the SSH key from github for the joe user
+   network:
+     - name: "Start k3s"
+       systemctl:
+         start: 
+         - k3s
+     - authorized_keys:
+     # Replace here with your ssh keys or github handle
+        joe: 
+        - github:mudler
+```
+
+Done! We are now ready to build the Docker container.
+
+The file structure should be like the following:
+
+```
+$> tree ./derivative
```

Comment:
4 files only, couldn't reduce it more! Actually if no ISO is needed, it's just 3 :smiley: